composer/datasets/streaming/dataset.py

```diff
@@ -392,6 +327,22 @@ def __iter__(self) -> Iterator[Any]:
             Iterator[Any]: Each sample.
         """
         Thread(target=self.download, daemon=True).start()
-
-        for idx in self._iter_ids():
-            yield self[idx]
+        num_workers = dist.get_local_world_size()
```

Comment:
Hmm this is probably not the right variable name to use, `num_workers` usually refers to the number of CPU workers, per device. Would recommend using `local_world_size` or something like `node_num_devices` for clarity.

Reply:
this makes sense

Reply:
In our offline conversation we opted to use `node_worker_id` and `node_num_workers` to hop between the samples.